src/pages/Recipe.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+
+import { Container, Row, Col, Table } from 'react-bootstrap';
+
+import queryString from 'query-string';
+import PropTypes from 'prop-types';
+import recipes from '../data/recipes.json';
+
+export default class Recipe extends Component {
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      flavors: []
+    };
+  }
+
+  componentDidMount() {
+    this.findRecipe();
+  }
+
+  findRecipe() {
+    const query = this.props.location.search;
+    const queryValues = queryString.parse(query);
+    const pageId = parseFloat(queryValues.id);
+
+    const recipe = recipes.filter(e => e.id === pageId)[0];
+
+    recipe.flavorTotal = 0;
+    recipe.flavors.forEach(flavor => {
+      recipe.flavorTotal += flavor.percent;
+    });
+
+    this.setState(recipe);
```

Comment:
Small change, but `this.setState({ recipe });` would be a little more future-proof in that we'll likely need to add _something_ else to the state and at that point tests might need to be updated if they are testing the component state.

Reply:
fixed